pkg/controller/replication_controller_test.go

```diff
@@ -149,29 +157,50 @@ func makeTestServer(t *testing.T, namespace, name string, podResponse, controlle
 	return httptest.NewServer(mux), &fakeUpdateHandler
 }
 
+// waitForPods polls the pod lister for timeout duration till numPods show up
+func waitForPods(podLister *cache.StoreToPodLister, numPods int, timeout time.Duration) error {
+	return wait.Poll(time.Millisecond, timeout, func() (bool, error) {
+		pods, err := podLister.List(labels.Everything())
+		if err != nil {
+			return false, err
+		}
+		return len(pods) == numPods, nil
+	})
+}
+
+func newStoreToPodLister(pods *api.PodList, store cache.Store) *cache.StoreToPodLister {
+	for _, p := range pods.Items {
+		store.Add(&p)
+	}
+	return &cache.StoreToPodLister{store}
+}
+
 func TestSyncReplicationControllerDoesNothing(t *testing.T) {
-	body, _ := latest.Codec.Encode(newPodList(2))
+	podList := newPodList(2, testLabels)
+	body, _ := latest.Codec.Encode(podList)
 	fakeHandler := util.FakeHandler{
 		StatusCode:   200,
 		ResponseBody: string(body),
 	}
 	testServer := httptest.NewServer(&fakeHandler)
 	defer testServer.Close()
-	client := client.NewOrDie(&client.Config{Host: testServer.URL, Version: testapi.Version()})
 
+	client := client.NewOrDie(&client.Config{Host: testServer.URL, Version: testapi.Version()})
 	fakePodControl := FakePodControl{}
-
 	manager := NewReplicationManager(client)
+	close(manager.stopCh)
+
 	manager.podControl = &fakePodControl
+	manager.podLister = newStoreToPodLister(podList, cache.NewStore(cache.MetaNamespaceKeyFunc))
 
 	controllerSpec := newReplicationController(2)
-
 	manager.syncReplicationController(controllerSpec)
 	validateSyncReplication(t, &fakePodControl, 0, 0)
 }
 
 func TestSyncReplicationControllerDeletes(t *testing.T) {
-	body, _ := latest.Codec.Encode(newPodList(2))
+	podList := newPodList(2, testLabels)
+	body, _ := latest.Codec.Encode(podList)
```

Comment:
Probably I'm not clear here :) In this test you are not using fakeHandler anywhere (at least that's my understanding). I'm just suggesting to delete it.

In other tests we have fakeUpdateHandler to verify whether Status is properly updated, but that's a different story and I'm not asking to delete it.